modules/contact-form/js/gutenblock.jsx

```diff
@@ -0,0 +1,568 @@
+/*global jQuery _ wp*/
+/** @jsx wp.element.createElement */
+/** @format */
+
+/**
+ * External dependencies
+ */
+import {
+	PanelBody,
+	TextControl,
+	TextareaControl,
+	CheckboxControl,
+	ToggleControl,
+	IconButton
+} from '@wordpress/components';
+
+import {
+	registerBlockType,
+	createBlock
+} from '@wordpress/blocks';
+
+import {
+	InnerBlocks,
+	InspectorControls
+} from '@wordpress/editor';
+
+import {
+	Component,
+	Fragment
+} from '@wordpress/element';
+
+import {
+	__
+} from '@wordpress/i18n';
+
+
+
+/**
+ * Components:
+ */
+
+class GrunionForm extends Component {
+	render() {
+		return (
+			<Fragment>
+				<InspectorControls>
+					<PanelBody title={ __( 'Submission Details', 'jetpack' ) }>
+						<TextControl
+							label={ __( 'What would you like the subject of the email to be?' ) }
+							value={ this.props.subject }
+							onChange={ ( x )=>this.props.setAttributes( { subject: x } ) }
+						/>
+						<TextControl
+							label={ __( 'Which email address should we send the submissions to?' ) }
+							value={ this.props.to }
+							onChange={ ( x )=>this.props.setAttributes( { to: x } ) }
+						/>
+						<TextControl
+							label={ __( 'What should the label on the form’s submit button say?' ) }
+							value={ this.props.submit_button_text ? this.props.submit_button_text : __( 'Submit »' ) }
+							onChange={ ( x )=>this.props.setAttributes( { submit_button_text: x } ) }
+						/>
+					</PanelBody>
+				</InspectorControls>
+				<div className="grunion-form">
+					{this.props.children}
+					<TextControl
+						className="button button-primary button-default grunion-submit-button"
+						value={ this.props.submit_button_text ? this.props.submit_button_text : __( 'Submit »' ) }
+						onChange={ ( x )=>this.props.setAttributes( { submit_button_text: x } ) }
+					/>
+				</div>
+			</Fragment>
+		);
+	}
+}
+
+class GrunionFieldRequiredToggle extends Component {
+	render() {
+		return (
+			<ToggleControl
+				label={ __( 'Required' ) }
+				checked={ this.props.required }
+				onChange={ this.props.onChange }
+			/>
+		);
+	}
+}
+
+class GrunionFieldSettings extends Component {
+	render() {
+		return (
+			<InspectorControls>
+				<PanelBody title={ __( 'Field Settings', 'jetpack' ) }>
+					<GrunionFieldRequiredToggle
+						required={ this.props.required }
+						onChange={ ( x )=>this.props.setAttributes( { required: x } ) }
+					/>
+				</PanelBody>
+			</InspectorControls>
+		);
+	}
+}
+
+class GrunionFieldLabel extends Component {
```

Comment:
Many of these smaller components don't need to be a class. They can just be a function. Components really only need to be a class if they need lifecycle functions (e.g. `componentDidMount`), or if they need to define bound event handler functions (and sometimes not even then)

Reply:
I've changed them over to functions where possible.